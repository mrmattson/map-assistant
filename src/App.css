.App {
  /*text-align: center;*/
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 200px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.MapView-map {
  width: 100%;
  height: 800px;
}

.ShapeListItem {
  font-weight: bold;
  font-size: 16px;
  line-height: 2.0;
}

label {
  margin-left: 20px;
}

.latLonInput {
  width: 80px;
}

textarea.RawTextInput-rawTextTextarea {
  width: 100%;
  height: 300px;
}

.OutputData-textarea {
  width: 100%;
  height: 300px;
}
