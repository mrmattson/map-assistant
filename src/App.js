import React from 'react'

import './App.css'
import logo from './logo.svg'
import GeoShapes from './containers/GeoShapes'

// eslint-disable-next-line react/prefer-stateless-function
class App extends React.Component {
  render () {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Map Assistant</h2>
          <h4>Calculations at your service.</h4>
        </div>
        <div className="container">
          <GeoShapes />
        </div>
      </div>
    )
  }
}

export default App
