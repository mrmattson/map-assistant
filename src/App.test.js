/* eslint-env jest */

import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { createStore } from 'redux'

import App from './App'
import shapeMapApp from './reducers'

const store = createStore(shapeMapApp)

it('renders without crashing', () => {
  mount(
    <Provider store={store}>
      <App />
    </Provider>
  )
})

it('default state snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
