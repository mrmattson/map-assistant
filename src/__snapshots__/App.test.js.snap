exports[`test default state snapshot 1`] = `
<div
  className="App">
  <div
    className="App-header">
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg" />
    <h2>
      Map Assistant
    </h2>
    <h4>
      Calculations at your service.
    </h4>
  </div>
  <div
    className="container">
    <div
      className="GeoShapes">
      <div
        className="row">
        <div
          className="col-md-4">
          <div
            className="RawTextInput">
            <div
              className="form-group">
              <label
                className="control-label"
                htmlFor={undefined}>
                Text Input
              </label>
              <textarea
                className="RawTextInput-rawTextTextarea form-control"
                id={undefined}
                onChange={[Function]}
                type={undefined}
                value="" />
            </div>
            <button
              className="btn btn-lg btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button">
              Process Text
            </button>
          </div>
          <div>
            <ul
              className="ShapeEditList list-group" />
            <button
              className="btn btn-lg btn-primary"
              disabled={true}
              onClick={[Function]}
              type="button">
              Save Selected Shapes
            </button>
          </div>
          <ul
            className="ShapeList list-group" />
          <div
            className="OutputData">
            <h4>
              KML Output
            </h4>
            <textarea
              className="OutputData-textarea"
              readOnly={true}
              value="<?xml version=\"1.0\" encoding=\"UTF-8\"?><kml xmlns=\"http://www.opengis.net/kml/2.2\"><Document><name>Generated from GeoConverter</name><description>This data is generated from data input into Geo File Converter.</description></Document></kml>" />
          </div>
        </div>
        <div
          className="col-md-8">
          <div>
            <div
              className="MapView-map" />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
